xlators/features/tier/src/tier.c

```diff
@@ -0,0 +1,2141 @@
+/*
+ *   Copyright (c) 2021 Pavilion Data Systems, Inc. <http://www.pavilion.io>
+ *   This file is part of GlusterFS.
+ *
+ *   This file is licensed to you under your choice of the GNU Lesser
+ *   General Public License, version 3 or any later version (LGPLv3 or
+ *   later), or the GNU General Public License, version 2 (GPLv2), in all
+ *   cases as published by the Free Software Foundation.
+ */
+
+#include <glusterfs/glusterfs.h>
+#include <glusterfs/xlator.h>
+#include <glusterfs/defaults.h>
+#include <glusterfs/syscall.h>
```

Comment:
```
11:55:33 /usr/home/jenkins/root/workspace/gh_freebsd-smoke/xlators/features/tier/src/tier.c: In function 'tier_setxattr_cbk':
11:55:33 /usr/home/jenkins/root/workspace/gh_freebsd-smoke/xlators/features/tier/src/tier.c:968:52: error: 'ENODATA' undeclared (first use in this function); did you mean 'NO_DATA'?
11:55:33   968 |                 gf_msg(this->name, GF_LOG_WARNING, ENODATA, 0,
```
Looks like you need to add `#include <glusterfs/compat-errno.h>` here to pass the freebsd smoke.